packages/kbn-es/src/cluster.js

```diff
@@ -276,6 +277,10 @@ exports.Cluster = class Cluster {
     }
     this._stopCalled = true;
 
+    if (this._serverlessNodes?.length) {
+      return await stopServerlessCluster(this._log, this._serverlessNodes);
+    }
```

Comment:
This part is not working if `node scripts/functional_tests --config <path to serverless config>` is manually aborted (e.g. Ctrl+C)
I checked that it is working fine for stateful run, but I can't find the reason why `cleanup` is not properly executed here.

Reply:
After looking into #23441, it seems like `treeKillAsync(this._process.pid)` is for Windows platform only.
I played around `AbortController` and it did not work for me, but I'm not sure I did it right way. Personally I believe that stopping ES containers on script abortion would be great for consistency with stateful testing, but if we can't do it at least printing a message on starting a script again or forcing to stop container on script re-run (could be with extra flag) will make it easier.

Reply:
Thanks for suggestion @dmlemeshko I'll see if I can get this working.

Reply:
@dmlemeshko added [7265db7](https://github.com/elastic/kibana/pull/162673/commits/7265db79ac7c3b865419ec117726d1c07e845063). I was also having trouble with `AbortController` and picking up signals within the child process in FTR.

It might be easier to handle this if we upgrade `execa 4.0.2 => 7.2.0` because we can pass an `AbortController` during child process spawn. I gave it a try but was having issues since they converted to ESM and it seemed to be a bit out of scope for the PR.